src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -395,13 +395,14 @@ ASR::Module_t* load_module(Allocator &al, SymbolTable *symtab,
 
 ASR::symbol_t* import_from_module(Allocator &al, ASR::Module_t *m, SymbolTable *current_scope,
                 std::string mname, std::string cur_sym_name, std::string new_sym_name,
-                const Location &loc) {
+                const Location &loc, bool skip_current_scope_check=false) {
     ASR::symbol_t *t = m->m_symtab->resolve_symbol(cur_sym_name);
     if (!t) {
         throw SemanticError("The symbol '" + cur_sym_name + "' not found in the module '" + mname + "'",
                 loc);
     }
-    if (current_scope->get_scope().find(cur_sym_name) != current_scope->get_scope().end()) {
+    if (!skip_current_scope_check &&
+        current_scope->get_scope().find(cur_sym_name) != current_scope->get_scope().end()) {
```

Comment:
I think the check should be on `new_sym_name` as that's the mangled thing, correct?

Reply:
Yes, I think that's a typo and needs to be fixed. Thanks!